external/storm-cassandra/src/main/java/org/apache/storm/cassandra/bolt/BaseCassandraBolt.java

```diff
@@ -0,0 +1,193 @@
+import backtype.storm.tuple.Tuple;
+import backtype.storm.utils.TupleUtils;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.exceptions.NoHostAvailableException;
+import org.apache.storm.cassandra.BaseExecutionResultHandler;
+import org.apache.storm.cassandra.CassandraContext;
+import org.apache.storm.cassandra.ExecutionResultHandler;
+import org.apache.storm.cassandra.client.CassandraConf;
+import org.apache.storm.cassandra.client.SimpleClient;
+import org.apache.storm.cassandra.client.SimpleClientProvider;
+import org.apache.storm.cassandra.executor.AsyncExecutor;
+import org.apache.storm.cassandra.executor.AsyncExecutorProvider;
+import org.apache.storm.cassandra.executor.AsyncResultHandler;
+import org.apache.storm.cassandra.query.CQLStatementTupleMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A base cassandra bolt.
+ *
+ * Default {@link backtype.storm.topology.base.BaseRichBolt}
+ */
+public abstract class BaseCassandraBolt<T> extends BaseRichBolt {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BaseCassandraBolt.class);
+
+    protected OutputCollector outputCollector;
+    
+    protected SimpleClientProvider clientProvider;
+    protected SimpleClient client;
+    protected Session session;
+    protected Map stormConfig;
+
+    protected CassandraConf cassandraConfConfig;
+
+    private CQLStatementTupleMapper mapper;
+    private ExecutionResultHandler resultHandler;
+
+    transient private  Map<String, Fields> outputsFields = new HashMap<>();
+
+    /**
+     * Creates a new {@link CassandraWriterBolt} instance.
+     * @param mapper
+     */
+    public BaseCassandraBolt(CQLStatementTupleMapper mapper, SimpleClientProvider clientProvider) {
+        this.mapper = mapper;
+        this.clientProvider = clientProvider;
+    }
+    /**
+     * Creates a new {@link CassandraWriterBolt} instance.
+     * @param tupleMapper
+     */
+    public BaseCassandraBolt(CQLStatementTupleMapper tupleMapper) {
+        this(tupleMapper, new CassandraContext());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void prepare(Map stormConfig, TopologyContext topologyContext, OutputCollector outputCollector) {
+        this.outputCollector = outputCollector;
+        this.stormConfig = stormConfig;
+        this.cassandraConfConfig = new CassandraConf(stormConfig);
+        this.client = clientProvider.getClient(this.stormConfig);
+        try {
+            session = client.connect();
+        } catch (NoHostAvailableException e) {
+            outputCollector.reportError(e);
+        }
+    }
+
+    public BaseCassandraBolt withResultHandler(ExecutionResultHandler resultHandler) {
+        this.resultHandler = resultHandler;
+        return this;
+    }
+
+    public BaseCassandraBolt withOutputFields(Fields fields) {
+        this.outputsFields.put(null, fields);
```

Comment:
Not a good practice to pass null to a map. It seems this is passed to set fields for default stream. You should use backtype.storm.utils.Utils.DEFAULT_STREAM_ID as key instead of null.